src/React/Basic.purs

```diff
@@ -369,6 +375,53 @@ foreign import toReactComponent
   -> { render :: Self props state -> JSX | spec }
   -> ReactComponent { | jsProps }
 
+foreign import data ReactContext :: Type -> Type
+
+-- | Create a `ReactContext` given a default value. Use `provider` and `consumer`
+-- | to provide and consume context values. Alternatively, use `contextProvider`
+-- | and `contextConsumer` directly if a `ReactComponent` is required for interop.
+-- |
+-- | ```purs
+-- | render self =
+-- |   R.div_
+-- |   [ R.button
+-- |     { onClick: capture_ $ self.setState \s -> s { counter = s.counter + 1 }
+-- |     , children: [ R.text "Tick!" ]
+-- |     }
+-- |   , provider countContext self.state.counter
+-- |     [ consumer countContext \counter ->
+-- |        [ R.text $ "Ticks: " <> (show counter)
+-- |        ]
+-- |     ]
+-- |   ]
```

Comment:
:wave: I'm late to the party as this PR has been merged for quite some time, and maybe my following question should have been posted elsewhere. if that's wrong I do apologize :bow: 

I've tried using the context API but failed to create a context as it's actually an Effect. This example doesn't cover the creation part. Am I missing something?

Reply:
Hello 👋 

You have two options:

1. Use `unsafePerformEffect` at the module level to create your context. It bends the rules a tiny bit, but in a safe, predictable way.
1. Define your context during app initialization. This is a pattern I've been using more often lately. Component creation is actually effectful as well, as [noted here](https://github.com/lumihq/purescript-react-basic/blob/df358364ece1cd4760429afae0bcca0eb518eb03/src/React/Basic.purs#L111) and [explained more thoroughly here](https://github.com/spicydonuts/purescript-react-basic-hooks/issues/30#issuecomment-647048533). This is also why the newer hooks api [defines component creation as an effect](https://github.com/spicydonuts/purescript-react-basic-hooks/blob/main/src/React/Basic/Hooks.purs#L85). You can see a more complete example of this pattern used with context [here](https://github.com/spicydonuts/purescript-react-basic-hooks/blob/master/examples/context/src/Example.purs#L11).

Reply:
Sorry I missed the notification of your reply. Thank you very much, it confirms what I was suspecting.

I'll go with the first option as I don't have a purescript main entry point in my setup.

The second option is the most appealing obviously. My take on all of this would be that if you have an main entry-point you could totally avoid React context and simply use a MonadReader-like instead (assuming we don't care about interop I guess).

Thank you @spicydonuts 🙇